agent/agent_pool.go

```diff
@@ -51,13 +51,88 @@ func NewAgentPool(l *logger.Logger, m *metrics.Collector, c AgentPoolConfig) *Ag
 		conf:             c,
 		logger:           l,
 		metricsCollector: m,
-		apiClient:        NewAPIClient(l, c.APIClientConfig),
 	}
 }
 
+func (r *AgentPool) RegisterOnly() error {
+	r.logger.Info("Registering agent with Buildkite...")
+
+	registered, err := r.registerAgent(r.createAgentTemplate())
+	if err != nil {
+		return err
+	}
+
+	r.logger.Info("Successfully registered agent \"%s\" with tags [%s]", registered.Name,
+		strings.Join(registered.Tags, ", "))
+
+	// Create a prefixed logger for some context in concurrent output
+	l := r.logger.WithPrefix(registered.Name)
+
+	l.Debug("Ping interval: %ds", registered.PingInterval)
```

Comment:
What do you think about logging these as a serialised JSON object so they can be parsed with `jq` instead of a `grep` / `sed` / `cut` combo?

Reply:
That's a super interesting idea, I'd love to do that for all loggers and have a `--json` flag.